packages/node/src/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -41,23 +48,239 @@ describe("ClientRegistrar", () => {
   }
 
   describe("getClient", () => {
-    it("isn't implemented", async () => {
+    it("Fails if there is not registration endpoint", async () => {
+      // Sets up the mock-up for DCR
+      const { Issuer } = jest.requireMock("openid-client");
+      const mockedIssuerConfig = mockIssuerConfig({
+        registration_endpoint: undefined,
+      });
+      const mockedIssuer = {
+        metadata: mockedIssuerConfig,
+      };
+      Issuer.discover = jest.fn().mockResolvedValueOnce(mockedIssuer);
+
+      // Run the test
       const clientRegistrar = getClientRegistrar({
         storage: mockStorageUtility({}),
       });
-      const registrationUrl = "https://idp.com/register";
       await expect(
         clientRegistrar.getClient(
           {
             sessionId: "mySession",
             redirectUrl: "https://example.com",
           },
-          {
-            ...IssuerConfigFetcherFetchConfigResponse,
-            registrationEndpoint: registrationUrl,
-          }
+          IssuerConfigFetcherFetchConfigResponse
         )
-      ).rejects.toThrow();
+      ).rejects.toThrow(
+        "Dynamic client registration cannot be performed, because issuer does not have a registration endpoint"
+      );
+    });
+
+    it("retrieves client id and secret from storage if they are present", async () => {
```

Comment:
Should we be getting the Client Name (as a string) too...? Is this description too implementation-specific (e.g. maybe change to 'client registration details' or something instead)? But at least if we **_are_** going to be explicit about what the test does, be very explicit, e.g.:
```suggestion
    it("retrieves the OIDC 'client_id' and 'client_secret' from storage if they are present", async () => {
```

Reply:
Yes, the client name could be retrieved from memory. Fixed in 723613e09a9a80001374bf3a8be9f5c37cb62ad1